framework/contracts/native/version-control/src/queries.rs

```diff
@@ -260,822 +260,856 @@ fn filter_modules_by_namespace(
+                            is_suspended: false,
+                            modules: vec![],
+                        };
+                        Ok(to_json_binary(&resp).unwrap())
+                    }
+                    _ => panic!("unexpected message"),
+                }
+            })
+            .with_contract_item(
+                account.addr(),
+                account::state::ACCOUNT_ID,
+                &AccountId::local(1),
+            )
+            .with_contract_item(
+                account.addr(),
+                cw2::CONTRACT,
+                &cw2::ContractVersion {
+                    contract: abstract_std::ACCOUNT.to_owned(),
+                    version: contract::CONTRACT_VERSION.to_owned(),
+                },
+            )
+            .with_smart_handler(&other_account, move |msg| {
+                match from_json(msg).unwrap() {
+                    account::QueryMsg::Config {} => {
+                        let abstr = AbstractMockAddrs::new(mock_api.clone());
+                        let resp = account::ConfigResponse {
+                            version_control_address: abstr.version_control,
+                            module_factory_address: abstr.module_factory,
+                            account_id: TEST_OTHER_ACCOUNT_ID, // mock value, not used
+                            is_suspended: false,
+                            modules: vec![],
+                        };
+                        Ok(to_json_binary(&resp).unwrap())
+                    }
+                    _ => panic!("unexpected message"),
+                }
+            })
+            .with_contract_item(
+                &other_account,
+                account::state::ACCOUNT_ID,
+                &AccountId::local(2),
+            )
+            .with_contract_item(
+                &other_account,
+                cw2::CONTRACT,
+                &cw2::ContractVersion {
+                    contract: abstract_std::ACCOUNT.to_owned(),
+                    version: contract::CONTRACT_VERSION.to_owned(),
+                },
+            )
+            .with_owner(account.addr(), Some(&abstr.owner))
+            .with_owner(&other_account, Some(&other_owner))
+    }
+
+    fn mock_init(
+        deps: &mut OwnedDeps<MockStorage, MockApi, MockQuerier>,
+    ) -> VersionControlTestResult {
+        let abstr = AbstractMockAddrs::new(deps.api);
+        let info = message_info(&abstr.owner, &[]);
+        let admin = info.sender.to_string();
+
+        contract::instantiate(
+            deps.as_mut(),
+            mock_env(),
+            info,
+            InstantiateMsg {
+                admin,
+                security_disabled: Some(true),
+                namespace_registration_fee: None,
+            },
+        )?;
+
+        Ok(())
+    }
+
+    /// Initialize the version_control with admin as creator and test account
+    fn mock_init_with_account(deps: &mut MockDeps) -> VCResult {
```

Comment:
aren't these fns also present in other files? (commands.rs / contract.rs)
is it possible for them to use the same setup fns?

Reply:
Couldn't make `mock_init_with_account()` compatible with both tests. Don't see a reason to complicate tests structure by reusing `execute_as()`